dali/kernels/imgproc/paste/paste_gpu.h

```diff
@@ -174,8 +175,11 @@ void CreateSampleDescriptors(
           patch.patch_end[1] = scaled_x_to_x[x + 1];
           patch.in = nullptr;  // to be filled later
           patch.in_sample_idx = max_paste == -1 ? -1 : sample.inputs[max_paste].in_idx;
+          patch.in_batch_idx = max_paste == -1 ? -1 : sample.inputs[max_paste].batch_idx;
           patch.in_pitch[0] = 0;
-          patch.in_pitch[1] = max_paste == -1 ? -1 : in_shape[sample.inputs[max_paste].in_idx][1];
+          int mp_in_idx = sample.inputs[max_paste].in_idx;
+          int mp_batch_idx = sample.inputs[max_paste].batch_idx;
```

Comment:
Just a line above you are comparing the `max_paste == -1` and here you are indexing into inputs, unless you did offset the pointer by one for this case, this might be wrong.

Reply:
oops, right, it's an artifact of hoisting this up from the inline if expr. good catch!